test/integration/download_agreement_test.rb

```diff
@@ -20,6 +20,12 @@ class DownloadAgreementTest < ActionDispatch::IntegrationTest
     sign_in_and_update @user
   end
 
+  # we use after(:each) because after(:all) will not have @download_agreement in scope
+  after(:each) do
+    @download_agreement&.destroy
```

Comment:
I think it might be simpler/faster to call this:
```
DownloadAgreement.destroy_all
DownloadAcceptance.destroy_all
```

If nothing else, it skips callbacks and ensures empty collections, which is what we want.

Reply:
I thought about that -- but if we want our tests to be parallelizable, we want to avoid destroy_all, since that could destroy instances currently in use by other tests

Reply:
Fair point - though I don't think these two classes will impact other tests.  But your hunch is likely correct.  We'll have to refactor all `destroy_all` calls when we parallelize.